modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,758 @@
+    area[2][1] = level.area[2][1]
+    area[2][2] = level.area[2][2]
+    area[1][1] = area[1][1] +  at_player * 500
+    area[2][1] = area[2][1] +  at_player * 500
+    islands[player.name] = area
+    local left_overs = variables["surface"].find_entities_filtered {area= area}
+    for i, ent in ipairs(left_overs) do
+        if ent.name ~= "market" and ent.name ~= "steel-chest" then
+            ent.destroy()
+        end
+    end
+    local tiles = {}
+    for i,tile in ipairs(save["tiles"]) do
+        tiles[i] = tile
+        tiles[i].position.x = tiles[i].position.x +  at_player * 500
+    end
+    variables["surface"].set_tiles(tiles)
+    for i, entity in ipairs(save["entity"]) do
+        local name = entity[1]
+        local position = {}
+        position.x = entity[2].x
+        position.y = entity[2].y
+        local force = entity[3]
+        local minable = entity[4]
+        position.x = position.x + at_player * 500
+        if name == "market" then
+            markets[#markets + 1] = position
+        end
+        --[[
+        if name == "steel-chest" then
+            chests[#chests+1] = {entity,player.name}
+        end
+        ]]
+        local ent = variables["surface"].create_entity{name = name , position = position , force = force }
+        if name == "steel-chest" then
+            chests[#chests+1] = {ent,player.name}
+        end
+        ent.minable = minable
+    end
+    centers[player.name] = {}
+    centers[player.name].x = level.center.x
+    centers[player.name].y = level.center.y
+    centers[player.name].x =  centers[player.name].x + at_player * 500
+    variables.walls[player.name] = {}
+    local wal = variables["surface"].find_entities_filtered {name = "stone-wall", area= area}
+    for i,wall in ipairs(wal) do
+        local p = wall.position
+        variables.walls[player.name][p.x..','..p.y] = true
+    end
+    local center = centers[player.name]
+    player.teleport({center.x,center.y},level.surface)
+    game.print("tp")
+end
+
+
+local function level_save()
+    local level = variables.level
+    for x=level.area[1][1],level.area[2][1] do
+        for y = level.area[1][2],level.area[2][2] do
+            local tile = variables["surface"].get_tile(x,y)
+            local table = {
+                name = tile.name,
+                position = tile.position
+            }
+            save["tiles"][#save["tiles"]+1] = table
+        end
+    end
+
+    save["entity"] = variables["surface"].find_entities_filtered {area = level.area}
+    for i, entity in ipairs(save["entity"]) do
+        local name = entity.name
+        if name ~= "character" then
+            if name == "steel-chest" then
+                chests[#chests+1] = {entity,game.connected_players[1].name}
+            end
+            local position = entity.position
+            local force = entity.force
+            local minbale =  entity.minable
+            local table = {name,position,force,minbale}
```

Comment:
Good to see local variables but they arent needed here since they are only used once.
```suggestion
            local table = {name,entity.position, entity.force, entity.minable}
```